x/gov/client/utils/query_test.go

```diff
@@ -0,0 +1,139 @@
+package utils
+
+import (
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/context"
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/gov/types"
+	"github.com/stretchr/testify/require"
+)
+
+type txMock struct {
+	address sdk.AccAddress
+	msgNum  int
+}
+
+func (tx txMock) ValidateBasic() sdk.Error {
+	return nil
+}
+
+func (tx txMock) GetMsgs() (msgs []sdk.Msg) {
+	for i := 0; i < tx.msgNum; i++ {
+		msgs = append(msgs, types.NewMsgVote(tx.address, 0, types.OptionYes))
+	}
+	return
+}
+
+func makeQuerier(txs []sdk.Tx) TxQuerier {
+	return func(cliCtx context.CLIContext, events []string, page, limit int) (*sdk.SearchTxsResult, error) {
+		start, end := client.Paginate(len(txs), page, limit, 100)
+		if start < 0 || end < 0 {
+			return nil, errors.New("unexpected error in test")
```

Comment:
We really should not error here.

Reply:
i would prefer to notify develop that introduced changes are conflicting with test setup, but ok

Reply:
This is inline with the rest of pagination logic within the SDK.